crf/src/main/java/org/sagebase/crf/step/active/HeartRateSampleProcessor.kt

```diff
@@ -302,13 +303,19 @@ class HeartRateSampleProcessor @JvmOverloads constructor(val videoProcessorFrame
         // (because the 3 seconds is removed by checking for the isLensCovered flag), but to match his output
         // it is reproduced here for testing purposes.
         val drop = if (dropSeconds > 0) dropSeconds * samplingRate - 1 else 0
-        val x = input.drop(drop).map { if (it.isFinite()) it else 0.0 }
+
+        // get minimum value in the input window and offset each signale frame by the minimum value observed in the window
+        //        let minValue = x.min()
+        //        let minnedOutX = x.map({$0 - minValue!})
+        val xInput = input.drop(drop)
+        val minValue = xInput.min()!!
```

Comment:
This would be safer as:
` val xMin = x.min() ?: 0.0`

Reply:
True except if you get here with null then you've already failed. ;)